what4/src/What4/Serialize/SETokens.hs

```diff
@@ -0,0 +1,265 @@
+-- | Definition of the S-Expression tokens used to
+-- (de)serialize What4 expressions.
+
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module What4.Serialize.SETokens
+    ( Atom(..)
+    , string, ident, int, nat, bitvec, bool, real, float
+    , string', ident'
+    , printAtom
+    , printSExpr
+    , parseSExpr
+    )
+    where
+
+import qualified Data.Foldable as F
+import qualified Data.Parameterized.NatRepr as PN
+import qualified Data.SCargot as SC
+import qualified Data.SCargot.Comments as SC
+import qualified Data.SCargot.Repr as SC
+import qualified Data.SCargot.Repr.WellFormed as SC
+import           Data.Semigroup
+import qualified Data.Sequence as Seq
+import           Data.Text (Text)
+import qualified Data.Text as T
+import qualified LibBF as BF
+import           Numeric.Natural ( Natural )
+import qualified Text.Parsec as P
+import           Text.Parsec.Text ( Parser )
+import           Text.Printf ( printf )
+import           Data.Ratio
+
+import           Data.Parameterized.Some ( Some(..))
+import qualified What4.BaseTypes as W4
+
+import           Prelude
+
+data Atom =
+  AId Text
+  -- ^ An identifier.
+  | AStr (Some W4.StringInfoRepr) Text
+  -- ^ A prefix followed by a string literal
+  -- (.e.g, AStr "u" "Hello World" is serialize as `#u"Hello World"`).
+  | AInt Integer
+  -- ^ Integer (i.e., unbounded) literal.
+  | ANat Natural
+  -- ^ Natural (i.e., unbounded) literal
+  | AReal Rational
+  -- ^ Real (i.e., unbounded) literal.
+  | AFloat (Some W4.FloatPrecisionRepr) BF.BigFloat
+  -- ^ A floating point literal (with precision)
+  | ABV Int Integer
+  -- ^ Bitvector, width and then value.
+  | ABool Bool
+  -- ^ Boolean literal.
+  deriving (Show, Eq, Ord)
+
+type SExpr = SC.WellFormedSExpr Atom
+
+string :: Some W4.StringInfoRepr -> Text -> SExpr
+string strInfo str = SC.A $ AStr strInfo str
+
+string' :: Some W4.StringInfoRepr -> String -> SExpr
+string' strInfo str = SC.A $ AStr strInfo (T.pack str)
+
+-- | Lift an unquoted identifier.
+ident :: Text -> SExpr
+ident = SC.A . AId
+
+ident' :: String -> SExpr
+ident' = SC.A . AId . T.pack
+
+
+-- | Lift a quoted identifier.
+-- quoted :: String -> SExpr
+-- quoted = SC.A . AQuoted
```

Comment:
Similarly, can this `AQuoted`-related code be deleted?

Reply:
Done